packages/android_alarm_manager/test/android_alarm_manager_test.dart

```diff
@@ -0,0 +1,203 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:android_alarm_manager/android_alarm_manager.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  setUpAll(() {
+    AndroidAlarmManager.channel
+        .setMockMethodCallHandler((MethodCall call) => null);
+  });
+
+  test('${AndroidAlarmManager.initialize}', () async {
+    AndroidAlarmManager.channel
+        .setMockMethodCallHandler((MethodCall call) async {
+      assert(call.method == 'AlarmService.start');
+      return true;
+    });
+
+    final bool initialized = await AndroidAlarmManager.initialize();
+
+    expect(initialized, isTrue);
+  });
+
+  group('${AndroidAlarmManager.oneShotAt}', () {
+    test('validates input', () async {
+      final DateTime validTime = DateTime.utc(1993);
+      final int validId = 1;
+      final Function validCallback = (int id) => null;
+
+      // Callback should take a single int param.
+      expect(
+          () => AndroidAlarmManager.oneShotAt(
+              validTime, validId, ((String foo) => foo)),
+          throwsAssertionError);
+
+      // ID should be less than 32 bits.
```

Comment:
Is this an actual restriction for AlarmManager?

Reply:
I'm not sure if this is enforced in the Android SDK, but the Dart API for this plugin is already enforcing it. See https://github.com/flutter/plugins/blob/master/packages/android_alarm_manager/lib/android_alarm_manager.dart#L189. I could change that if it's wrong but I wanted this to completely avoid changing any real functionality and just test what was already available and document it as-is.

Reply:
Ah I didn't realize that. We should probably just leave it as-is for now then.